### PR TITLE
Fix DoFnTester side inputs

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

The side inputs were being stored as iterables, but being returned as
the raw type.

Store the side input values directly instead.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @kennknowles

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java [hunk 1]
```diff
@@ -124,29 +124,25 @@ public void setSideInputs(Map<PCollectionView<?>, Iterable<WindowedValue<?>>> si
...
-      Iterable<?> value) {
-    sideInputs.put(
-        sideInput,
-        Iterables.transform(value, new Function<Object, WindowedValue<?>>() {
-          @Override
-          public WindowedValue<?> apply(Object input) {
-            return WindowedValue.valueInGlobalWindow(input);
-          }
-        }));
+  public <T> void setSideInput(
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java]: Why this change? The additional method for just a single value seems fine, but the case for adding globally windowed data remains. At the least, update the javadoc to indicate that you aren't setting the entirety of the PCollection contents. Is it unused?

Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java]: I've done a bit of cleanup, including removing the setSideInput without window method.

The other two commits are mainly cleanup of dead (or effectively dead) code.

**Comment by kennknowles** [comment]
LGTM. Nice catch. I'll merge in a couple minutes, let Travis have a chance to catch up.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java [hunk 2]
```diff
@@ -103,62 +101,33 @@
...
-   * <p>If needed, first creates a fresh instance of the DoFn under test.
+   * <p>The provided value is the final value of the side input in the specified window, not
+   * the value of the input PCollection in that window.
    *
-   * <p>If this isn't called, {@code DoFnTester} assumes the
-   * {@code DoFn} doesn't emit to any side outputs.
+   * <p>If this isn't called, {@code DoFnTester} will return the default value for any side input
+   * that is used.
    */
-  public void setSideOutputTags(TupleTagList sideOutputTags) {
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java]: ...

Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java]: This line is mostly a diff failure.

I believe that both Side Input behaviors are reasonable; although I can also see an argument for throwing an illegal state exception if the `(PCollectionView, BoundedWindow)` was not specified, and adding a `setSideInputDefault(PCollectionView, BoundedWindow)` method.
